com.unity.render-pipelines.universal/Documentation~/writing-shaders-urp-basic-unlit-structure.md

```diff
@@ -0,0 +1,177 @@
+            };
+
+            struct Varyings
+            {
+                // The positions in this struct must have the SV_POSITION semantic.
+                float4 positionHCS  : SV_POSITION;
+            };            
+
+            // The vertex shader definition with properties defined in the Varyings 
+            // structure. The type of the vert function must match the type (struct)
+            // that it returns.
+            Varyings vert(Attributes IN)
+            {
+                // Declaring the output object (OUT) with the Varyings struct.
+                Varyings OUT;
+                // The TransformObjectToHClip function transforms vertex positions
+                // from object space to homogenous space
+                OUT.positionHCS = TransformObjectToHClip(IN.positionOS.xyz);
+                // Returning the output.
+                return OUT;
+            }
+
+            // The fragment shader definition.
+            half4 frag() : SV_Target
+            {
+                // Defining the color variable and returning it.
+                half4 customColor;
+                customColor = half4(0.5, 0, 0, 1);
+                return customColor;
+            }
+            ENDHLSL
+        }
+    }
+}
+```
+
+The shader paints the GameObject dark red (RGB value (0.5, 0, 0)).
+
+![The shader paints the GameObject dark red](Images/shader-examples/unlit-shader-tutorial-basic-hardcoded-color.jpg)
+
+The following section introduces you to the structure of this basic shader.
+
+<a name="basic-shaderlab-structure"></a>
+
+## Basic ShaderLab structure
+
+Unity shader assets are written in a Unity-specific language called [ShaderLab](https://docs.unity3d.com/Manual/SL-Shader.html). 
+
+The shader in this example has the following blocks:
+
+* [Shader](#shader)
+* [Properties](#properties)
+* [SubShader](#subshader)
+* [Pass](#pass)
+* [HLSLPROGRAM](#hlsl)
+
+<a name="shader"></a>
+
+### Shader block
+
+A ShaderLab file starts with the `Shader` declaration.
+
+```c++
+Shader "Example/URPUnlitShaderBasic"
+```
+
+The path in this declaration determines the location of the shader in the Shader menu on a Material.
+
+![location of the shader in the Shader menu on a Material](Images/shader-examples/urp-material-ui-shader-path.png)
+
+<a name="properties"></a>
+
+### Properties block
+
+The [Properties](https://docs.unity3d.com/Manual/SL-Properties.html) block contains the declarations of properties that users can set in the Inspector window on a Material.
+
+In this example, the Properties block is empty, since this shader does not expose any Material properties that a user can define. TODO:reference to Color. 
+
+### SubShader block
+
+A ShaderLab file contains one or more [SubShader](https://docs.unity3d.com/Manual/SL-SubShader.html) blocks. When rendering a mesh, Unity selects the first SubShader block that is compatible with the GPU on the target device.
+
+A SubShader block contains the __Tags__ element.
```

Comment:
A SubShader block can optionally contain a SubShader Tags block. You use the Tags keyword to declare a SubShader Tags block.